osu.Game/Screens/Play/Player.cs

```diff
@@ -50,6 +50,8 @@ public class Player : ScreenWithBeatmapBackground
 
         public override bool HideOverlaysOnEnter => true;
 
+        public override bool AllowBlockedGlobalActions => true;
```

Comment:
might have to move this elsewhere but unsure for now

Reply:
if disabling certain global action in-game is the way to go, then judging by the name of this property alone, this shouldn't really exist anywhere, sounds like you just want to attach an `IKeyBindingHandler<GlobalAction>` to `Player` and swallow them.

that way you don't have to complicate `OsuGame` and weirdly (talking from someone who just reads that class) allowing `ResetInputSettings` and `ToggleGameplayMouseButtons` but blocking anything else.

Reply:
Oh wow I didn't even think of that. Will rewrite and push now.

Reply:
And I just noticed that this doesn't work as intended after I've already pushed

Reply:
which probably wouldn't have happened if you had added tests for this. please do so.

Reply:
If the goal is to disable potential activation of overlays during gameplay, wouldn't it be better to use the `InitialOverlayActivation` property (see in `OsuScreen`), and possibly the game level bindable `OverlayActivationMode` to re-enable use of overlays when gameplay is paused?